### PR TITLE
Move Computeschedule from private to public preview

### PR DESCRIPTION
**Pull request by NneOkeke**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    - Move private preview API to public preview
    -  Other PR reviews of this RP in the private repo [Initial commit merged](https://github.com/Azure/azure-rest-api-specs-pr/pull/15887), [new version merged](https://github.com/Azure/azure-rest-api-specs-pr/pull/17381), [struct fix merged](https://github.com/Azure/azure-rest-api-specs-pr/pull/18083)

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
Please fix the spellCheck failures by using the appropriate camelCase or PascalCase naming styles in each context, and/or using the common-types LocationParameter definition.

**Comment by TimLovellSmith** [comment]
Please fix avocado warnings about unreferenced examples:
https://github.com/Azure/azure-rest-api-specs/pull/29660/checks?check_run_id=26913650209

**Comment by TimLovellSmith** [comment]
Please fix avocado warnings about readme file not including all APIs/jsons in the default tag, or reply with justifications for e.g. deprecated APIs

File: specification/computeschedule/ComputeSchedule.Management/tspconfig.yaml
Comment (TimLovellSmith) [inline specification/computeschedule/ComputeSchedule.Management/tspconfig.yaml:11]: per typespec linter, please change to extends:
"@azure-tools/typespec-azure-rulesets/resource-manager"

**Comment by TimLovellSmith** [comment]
Per above comments, please look into the tool validation errors and then remove ARMChangesRequested tag, to unblock the actual manual ARM review.

**Comment by NneOkeke** [comment]
> Please fix avocado warnings about readme file not including all APIs/jsons in the default tag, or reply with justifications for e.g. deprecated APIs

Hello, the files in question are from a deprecated API version.

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (madhura-np) [inline specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:880]: >type [](http://example.com/codeflow?start=11&length=4)

is this an ArmId always? If so please add "format": "armId"

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (madhura-np) [inline specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:908]: >resourceId [](http://example.com/codeflow?start=9&length=10)

same as above with the format.

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (madhura-np) [inline specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:950]: >subscriptionId [](http://example.com/codeflow?start=9&length=14)

"format": "uuid" ?

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (NneOkeke) [reply specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:848]: Quick question on this, will this require a new version since this is a breaking change according to [this document](https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7B0D01EC65-C772-4A6E-9326-0279E586520C%7D&file=RestAPI_BreakingChangeGuide.docx&action=default&mobileredirect=true&DefaultItemOpen=1&share=IQFl7AENcsduSpMmAnnlhlIMAWyykwWAcYcItf7Ewp6Z6qY)?

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (NneOkeke) [reply specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:925]: Using the uuid made the validation for uuid fail, for now, I'd like to keep these as strings.

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (NneOkeke) [reply specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:856]: changed this format to armid

File: specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json
Comment (NneOkeke) [reply specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json:882]: Changed this format to armid

**Comment by raosuhas** [comment]
Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?

**Comment by NneOkeke** [comment]
> Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?

I am adding one api version to the public repo, the `2024-06-01-preview` version, [this](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/computeschedule) is the merged version in the private repo, it has gone through review and sign-off in the private repo. 

Would it make it easier if I remove the other versions from this particular PR?

**Comment by raosuhas** [comment]
> > Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?
> 
> I am adding one api version to the public repo, the `2024-06-01-preview` version, [this](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/computeschedule) is the merged version in the private repo, it has gone through review and sign-off in the private repo.
> 
> Would it make it easier if I remove the other versions from this particular PR?

Yeah it would be easier if we just have one PR per version. And If the changes are no different from the private repo then we can sign off on them much easier

**Comment by NneOkeke** [comment]
> > > Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?
> > 
> > 
> > I am adding one api version to the public repo, the `2024-06-01-preview` version, [this](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/computeschedule) is the merged version in the private repo, it has gone through review and sign-off in the private repo.
> > Would it make it easier if I remove the other versions from this particular PR?
> 
> Yeah it would be easier if we just have one PR per version. And If the changes are no different from the private repo then we can sign off on them much easier

I've updated this PR to reflect only the version we want to release.

**Comment by raosuhas** [comment]
> > > > Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?
> > > 
> > > 
> > > I am adding one api version to the public repo, the `2024-06-01-preview` version, [this](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/computeschedule) is the merged version in the private repo, it has gone through review and sign-off in the private repo.
> > > Would it make it easier if I remove the other versions from this particular PR?
> > 
> > 
> > Yeah it would be easier if we just have one PR per version. And If the changes are no different from the private repo then we can sign off on them much easier
> 
> I've updated this PR to reflect only the version we want to release.

Thanks ..

The 2024-06-01-preview is this any different from the one in the private repo ? What are the changes that need to be reviewed that are different from the private repo ?

**Comment by NneOkeke** [comment]
> > > > > Can you please explain a bit more about this PR ? Are you adding the 3 different api versions ? Are these aapi versions already reviewed and signed off in the private repo ?
> > > > 
> > > > 
> > > > I am adding one api version to the public repo, the `2024-06-01-preview` version, [this](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/computeschedule) is the merged version in the private repo, it has gone through review and sign-off in the private repo.
> > > > Would it make it easier if I remove the other versions from this particular PR?
> > > 
> > > 
> > > Yeah it would be easier if we just have one PR per version. And If the changes are no different from the private repo then we can sign off on them much easier
> > 
> > 
> > I've updated this PR to reflect only the version we want to release.
> 
> Thanks ..
> 
> The 2024-06-01-preview is this any different from the one in the private repo ? What are the changes that need to be reviewed that are different from the private repo ?

This `2024-06-01-preview` version isn't different from what is in the private repo, they're the same spec files. This PR is to move the private specs to the public repo for public preview release.

**Comment by razvanbadea-msft** [comment]
Comparing this version with the one from the private repo, you are missing some examples files (10 vs 18)

**Review by razvanbadea-msft — APPROVED** [review]

**Pull request merged**
